subsys/net/lib/dns/dns_cache.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2024 Endress+Hauser AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/dns_resolve.h>
+#include "dns_cache.h"
+
+LOG_MODULE_REGISTER(net_dns_cache, CONFIG_DNS_RESOLVER_LOG_LEVEL);
+
+static void dns_cache_clean(struct dns_cache const *cache);
+
+int dns_cache_flush(struct dns_cache *cache)
+{
+	k_mutex_lock(cache->lock, K_FOREVER);
+	for (size_t i = 0; i < cache->size; i++) {
+		cache->entries[i].in_use = false;
+	}
+	k_mutex_unlock(cache->lock);
+
+	return 0;
+}
+
+int dns_cache_add(struct dns_cache *cache, char const *query, struct dns_addrinfo const *addrinfo,
+		  uint32_t ttl)
+{
+	int64_t oldest_entry_uptime = INT64_MAX;
+	size_t index_to_replace = 0;
+	bool found_empty = false;
+
+	if (cache == NULL || query == NULL || addrinfo == NULL || ttl == 0) {
+		return -EINVAL;
+	}
+
+	if (strlen(query) >= DNS_MAX_NAME_LEN) {
+		NET_WARN("Query string to big to be processed %u >= DNS_MAX_NAME_LEN",
```

Comment:
If we set the max length behind the kconfig option, then we need to print the value here instead of the name.

Reply:
Adjusted